diff -ur qgis-2.0.1_orig/src/plugins/CMakeLists.txt qgis-2.0.1/src/plugins/CMakeLists.txt
--- qgis-2.0.1_orig/src/plugins/CMakeLists.txt	2013-09-11 06:06:52.000000000 +0700
+++ qgis-2.0.1/src/plugins/CMakeLists.txt	2013-10-08 17:27:13.299725892 +0700
@@ -15,7 +15,10 @@
 ADD_SUBDIRECTORY(georeferencer)
 ADD_SUBDIRECTORY(gps_importer)
 ADD_SUBDIRECTORY(topology)
-ADD_SUBDIRECTORY(offline_editing)
+
+IF (SPATIALITE_FOUND)
+  ADD_SUBDIRECTORY(offline_editing)
+ENDIF (SPATIALITE_FOUND)
 
 IF (POSTGRES_FOUND)
   ADD_SUBDIRECTORY(spit)
