diff -ur qgis-2.0.1_orig/python/CMakeLists.txt qgis-2.0.1/python/CMakeLists.txt
--- qgis-2.0.1_orig/python/CMakeLists.txt	2013-10-11 12:45:31.000000000 +0700
+++ qgis-2.0.1/python/CMakeLists.txt	2013-10-11 12:46:52.690905715 +0700
@@ -91,6 +91,12 @@
 FILE(GLOB_RECURSE sip_files_gui gui/*.sip)
 SET(SIP_EXTRA_FILES_DEPEND ${sip_files_core} ${sip_files_gui})
 SET(SIP_EXTRA_OPTIONS ${PYQT4_SIP_FLAGS} -o -a ${CMAKE_BINARY_DIR}/python/qgis.gui.api)
+
+IF(UNIX AND NOT SIP_VERSION_NUM LESS 265984)
+  SET(SIP_EXTRA_OPTIONS -P ${SIP_EXTRA_OPTIONS})
+  ADD_DEFINITIONS(-Dprotected=public)
+ENDIF(UNIX AND NOT SIP_VERSION_NUM LESS 265984)
+
 ADD_SIP_PYTHON_MODULE(qgis.gui gui/gui.sip qgis_core qgis_gui)
 
 # additional analysis includes
