diff -Nurp cairo-dock-2.4.0~2.old/src/gldit/cairo-dock-keybinder.h cairo-dock-2.4.0~2/src/gldit/cairo-dock-keybinder.h
--- cairo-dock-2.4.0~2.old/src/gldit/cairo-dock-keybinder.h	2012-04-20 18:38:10.657711505 +0200
+++ cairo-dock-2.4.0~2/src/gldit/cairo-dock-keybinder.h	2012-04-20 18:39:32.806152151 +0200
@@ -32,7 +32,7 @@
 #ifndef __CD_KEY_BINDER_H__
 #define __CD_KEY_BINDER_H__
 
-#include <glib/gtypes.h>
+#include <glib.h>
 #include "cairo-dock-struct.h"
 #include "cairo-dock-manager.h"
 
