--- work.orig/vmci-only/linux/driver.c	2013-10-18 17:56:11.000000000 +0000
+++ work/vmci-only/linux/driver.c	2013-11-14 16:54:17.621868821 +0000
@@ -737,7 +737,7 @@ LinuxDriver_Ioctl(struct inode *inode,
          goto init_release;
       }
 
-      user = current_uid();
+      user = __kuid_val(current_uid());
       retval = VMCIContext_InitContext(initBlock.cid, initBlock.flags,
                                        0 /* Unused */, vmciLinux->userVersion,
                                        &user, &vmciLinux->context);

