diff -Nurp work.orig/vsock-only/linux/af_vsock.c work/vsock-only/linux/af_vsock.c
--- work.orig/vsock-only/linux/af_vsock.c	2013-11-14 17:01:51.345452830 +0000
+++ work/vsock-only/linux/af_vsock.c	2013-11-14 17:00:49.187701789 +0000
@@ -2869,7 +2869,7 @@ __VSockVmciCreate(struct net *net,
       vsk->connectTimeout = psk->connectTimeout;
    } else {
       vsk->trusted = capable(CAP_NET_ADMIN);
-      vsk->owner = current_uid();
+      vsk->owner = __kuid_val(current_uid());
       vsk->queuePairSize = VSOCK_DEFAULT_QP_SIZE;
       vsk->queuePairMinSize = VSOCK_DEFAULT_QP_SIZE_MIN;
       vsk->queuePairMaxSize = VSOCK_DEFAULT_QP_SIZE_MAX;

