diff --git a/test/rbkt/CMakeLists.txt b/test/rbkt/CMakeLists.txt
index da004e2..b6b9211 100644
--- a/test/rbkt/CMakeLists.txt
+++ b/test/rbkt/CMakeLists.txt
@@ -75,7 +75,7 @@ ELSE (NOT Boost_FILESYSTEM_FOUND)
 
 ENDIF (NOT Boost_FILESYSTEM_FOUND)
 
-IF (ZORBATEST_TESTDRIVER_MT_AVAILABLE)
+IF (FALSE)
   INCLUDE_DIRECTORIES(${Boost_INCLUDE_DIRS})
   LINK_DIRECTORIES(${Boost_LIBRARY_DIRS})
 
@@ -102,7 +102,7 @@ IF (ZORBATEST_TESTDRIVER_MT_AVAILABLE)
     "If true, use testdriver_mt (when available) for running XQTS")
   MARK_AS_ADVANCED (ZORBATEST_USE_MT_XQTS)
 
-ENDIF (ZORBATEST_TESTDRIVER_MT_AVAILABLE)
+ENDIF (FALSE)
 
 INCLUDE(ZorbaAddTest)
 
