--- xfce4-radio-plugin-0.5.1.orig/configure.in
+++ xfce4-radio-plugin-0.5.1/configure.in
@@ -19,6 +19,8 @@ RADIO_PLUGIN_VERSION=radio_plugin_versio
 AM_INIT_AUTOMAKE([xfce4-radio-plugin], [$RADIO_PLUGIN_VERSION])
 AM_CONFIG_HEADER([config.h])
 
+AC_CHECK_LIB(m, ceilf, LIBM="-lm", AC_MSG_ERROR(Missing working libm math lib))
+
 AM_MAINTAINER_MODE
 
 dnl Check for UNIX variants
--- xfce4-radio-plugin-0.5.1.orig/panel-plugin/Makefile.am
+++ xfce4-radio-plugin-0.5.1/panel-plugin/Makefile.am
@@ -1,6 +1,8 @@
 plugindir = $(libexecdir)/xfce4/panel-plugins
 plugin_PROGRAMS = xfce4-radio-plugin
 
+AM_LDFLAGS = -lm
+
 xfce4_radio_plugin_SOURCES =						\
 	xfce4-radio.c xfce4-radio.h					\
 	radio.c radio.h							\

