commit 22fd8f18e627f9526bc2e9cac48241b115fe8c62
Author: J. García <jyo.garcia@yahoo.com>
Date:   Sun Oct 18 11:44:58 2015 -0600

    change linker flags for utils library

diff --git a/src/utils/Makefile.am b/src/utils/Makefile.am
index cea0662..5784006 100644
--- a/src/utils/Makefile.am
+++ b/src/utils/Makefile.am
@@ -1,6 +1,6 @@
 lib_LTLIBRARIES = libbd_utils.la
 libbd_utils_la_CFLAGS = $(GLIB_CFLAGS) -Wall -Wextra -Werror
-libbd_utils_la_LDFLAGS = -version-info 0:1:0
+libbd_utils_la_LDFLAGS = -lm -version-info 0:1:0
 libbd_utils_la_LIBADD = $(GLIB_LIBS)
 libbd_utils_la_SOURCES = utils.h exec.c exec.h sizes.c sizes.h
 
