From 53933e21316db15d70b540501af716860c0ff055 Mon Sep 17 00:00:00 2001
From: Ondrej Holy <oholy@redhat.com>
Date: Fri, 15 Nov 2013 17:32:04 +0100
Subject: [PATCH 09/11] sftp: missing enumeration value warning fix

---
 daemon/gvfsbackendsftp.c | 60 +++++++++++++++++++++---------------------------
 1 file changed, 26 insertions(+), 34 deletions(-)

diff --git a/daemon/gvfsbackendsftp.c b/daemon/gvfsbackendsftp.c
index 17ba4da..a95d908 100644
--- a/daemon/gvfsbackendsftp.c
+++ b/daemon/gvfsbackendsftp.c
@@ -2564,31 +2564,27 @@ try_seek_on_read (GVfsBackend *backend,
   GVfsBackendSftp *op_backend = G_VFS_BACKEND_SFTP (backend);
   GDataOutputStream *command;
 
-  if (job->seek_type == G_SEEK_END)
+  switch (job->seek_type)
     {
+    case G_SEEK_CUR:
+      handle->offset += job->requested_offset;
+      break;
+    case G_SEEK_SET:
+      handle->offset = job->requested_offset;
+      break;
+    case G_SEEK_END:
       command = new_command_stream (op_backend,
                                     SSH_FXP_FSTAT);
       put_data_buffer (command, handle->raw_handle);
       queue_command_stream_and_free (op_backend, command, seek_read_fstat_reply, G_VFS_JOB (job), handle);
+      return TRUE;
     }
-  else
-    {
-      switch (job->seek_type)
-        {
-        case G_SEEK_CUR:
-          handle->offset += job->requested_offset;
-          break;
-        case G_SEEK_SET:
-          handle->offset = job->requested_offset;
-          break;
-        }
 
-      if (handle->offset < 0)
-        handle->offset = 0;
+  if (handle->offset < 0)
+    handle->offset = 0;
 
-      g_vfs_job_seek_read_set_offset (job, handle->offset);
-      g_vfs_job_succeeded (G_VFS_JOB (job));
-    }
+  g_vfs_job_seek_read_set_offset (job, handle->offset);
+  g_vfs_job_succeeded (G_VFS_JOB (job));
 
   return TRUE;
 }
@@ -3695,31 +3691,27 @@ try_seek_on_write (GVfsBackend *backend,
   GVfsBackendSftp *op_backend = G_VFS_BACKEND_SFTP (backend);
   GDataOutputStream *command;
 
-  if (job->seek_type == G_SEEK_END)
+  switch (job->seek_type)
     {
+    case G_SEEK_CUR:
+      handle->offset += job->requested_offset;
+      break;
+    case G_SEEK_SET:
+      handle->offset = job->requested_offset;
+      break;
+    case G_SEEK_END:
       command = new_command_stream (op_backend,
                                     SSH_FXP_FSTAT);
       put_data_buffer (command, handle->raw_handle);
       queue_command_stream_and_free (op_backend, command, seek_write_fstat_reply, G_VFS_JOB (job), handle);
+      return TRUE;
     }
-  else
-    {
-      switch (job->seek_type)
-        {
-        case G_SEEK_CUR:
-          handle->offset += job->requested_offset;
-          break;
-        case G_SEEK_SET:
-          handle->offset = job->requested_offset;
-          break;
-        }
 
-      if (handle->offset < 0)
-        handle->offset = 0;
+  if (handle->offset < 0)
+    handle->offset = 0;
 
-      g_vfs_job_seek_write_set_offset (job, handle->offset);
-      g_vfs_job_succeeded (G_VFS_JOB (job));
-    }
+  g_vfs_job_seek_write_set_offset (job, handle->offset);
+  g_vfs_job_succeeded (G_VFS_JOB (job));
 
   return TRUE;
 }
-- 
1.9.0

