Index: noad-0.7.2/configure.ac
===================================================================
--- noad-0.7.2.orig/configure.ac
+++ noad-0.7.2/configure.ac
@@ -267,7 +267,6 @@ AC_FUNC_STRFTIME
 AC_CHECK_FUNCS([gettimeofday localtime_r memmove memset mkdir select socket strchr strdup strerror strrchr strstr])
 AC_CHECK_FUNCS([strsignal])
 #CXXFLAGS='-g -O3 -pg'
-CXXFLAGS='-g -O3'
 
 dnl ---------------------------------------------------------------------------
 dnl check for a usable avcodec.h 
