diff -up EMBOSS-6.4.0/ajax/Makefile.am.system-hack EMBOSS-6.4.0/ajax/Makefile.am
--- EMBOSS-6.4.0/ajax/Makefile.am.system-hack	2011-11-23 17:14:51.094999577 +0100
+++ EMBOSS-6.4.0/ajax/Makefile.am	2011-11-23 17:14:51.116999682 +0100
@@ -1,6 +1,4 @@
 ## Process this file with automake to produce Makefile.in
-if !ESYSTEMLIBS
-EXTRA_DIRS = expat zlib pcre
-endif
+EXTRA_DIRS = 
 
 SUBDIRS = $(EXTRA_DIRS) core graphics ensembl ajaxdb acd
