diff -up EMBOSS-6.4.0/configure.in.system-zlib EMBOSS-6.4.0/configure.in
--- EMBOSS-6.4.0/configure.in.system-zlib	2011-11-23 17:07:55.000000000 +0100
+++ EMBOSS-6.4.0/configure.in	2011-11-23 17:08:36.848586027 +0100
@@ -644,6 +644,7 @@ if test "${enable_systemlibs}" = "yes" ;
   have_systemlibs=yes
   AC_CHECK_LIB(pcre, pcre_compile)
   AC_CHECK_LIB(expat, XML_ParserCreate)
+  AC_CHECK_LIB(z, compress)
 fi
 AM_CONDITIONAL([ESYSTEMLIBS], [test "$have_systemlibs" = "yes"])
 AC_SUBST([ESYSTEMLIBS])
@@ -808,7 +809,6 @@ AC_CONFIG_FILES([Makefile
                  ajax/core/Makefile
                  ajax/ensembl/Makefile
                  ajax/graphics/Makefile
-                 ajax/zlib/Makefile
                  doc/Makefile
                  doc/manuals/Makefile
                  doc/programs/Makefile
diff -up EMBOSS-6.4.0/configure.system-zlib EMBOSS-6.4.0/configure
