diff --git a/babl/Makefile.am b/babl/Makefile.am
index 35a4fdb..f85eb37 100644
--- a/babl/Makefile.am
+++ b/babl/Makefile.am
@@ -95,7 +95,8 @@ INTROSPECTION_COMPILER_ARGS = --includedir=$(srcdir)
 if HAVE_INTROSPECTION
 Babl-$(BABL_API_VERSION).gir: $(G_IR_SCANNER) $(library_include_HEADERS) $(c_sources) $(srcdir)/Makefile.am
 	$(INTROSPECTION_SCANNER) -v --namespace Babl --nsversion=$(BABL_API_VERSION) \
-            --add-include-path=$(srcdir) --add-include-path=. \
+            --identifier-prefix= --symbol-prefix=babl \
+	    --add-include-path=$(srcdir) --add-include-path=. \
             --library=libbabl-$(BABL_API_VERSION).la \
             --libtool="$(LIBTOOL)" \
             --output $@ \
