diff --git a/retroshare-gui/src/RetroShare.pro b/retroshare-gui/src/RetroShare.pro
index 0099eb0..f6343ce 100644
--- a/retroshare-gui/src/RetroShare.pro
+++ b/retroshare-gui/src/RetroShare.pro
@@ -1,4 +1,5 @@
 CONFIG += qt gui uic qrc resources uitools idle bitdht # framecatcher# blogs
+CONFIG += install_rs
 QT     += network xml script 
 
 TEMPLATE = app
diff --git a/retroshare-nogui/src/retroshare-nogui.pro b/retroshare-nogui/src/retroshare-nogui.pro
index b52c9e4..fcb4abf 100644
--- a/retroshare-nogui/src/retroshare-nogui.pro
+++ b/retroshare-nogui/src/retroshare-nogui.pro
@@ -1,6 +1,7 @@
 TEMPLATE = app
 TARGET = retroshare-nogui
 CONFIG += bitdht
+CONFIG += install_rs
 #CONFIG += introserver
 
 ################################# Linux ##########################################
@@ -18,6 +19,12 @@ linux-* {
 	LIBS += -lssl -lgpgme -lupnp -lixml -lgnome-keyring
 }
 
+install_rs {
+        INSTALLS += binary_rs
+        binary_rs.path = $$(PREFIX)/usr/bin
+        binary_rs.files = ./retroshare-nogui
+}
+
 linux-g++ {
 	OBJECTS_DIR = temp/linux-g++/obj
 }
