From 1aba86e149b8418bf6dee6c516d110f8f3c75138 Mon Sep 17 00:00:00 2001
From: Antti Ajanki <antti.ajanki@iki.fi>
Date: Sat, 25 May 2013 18:33:43 +0300
Subject: [PATCH 048/110] [XMLBase] Handle comment, PI and CData nodes in
 buildCopy()

---
 src/backends/xml_support.cpp | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/src/backends/xml_support.cpp b/src/backends/xml_support.cpp
index a55e858..e0d54ba 100644
--- a/src/backends/xml_support.cpp
+++ b/src/backends/xml_support.cpp
@@ -139,11 +139,21 @@ void XMLBase::addDefaultNamespaceRecursive(xmlNodePtr node, xmlNsPtr ns)
 
 xmlpp::Node* XMLBase::buildCopy(const xmlpp::Node* src)
 {
+	const xmlpp::ContentNode* contentnode;
 	const xmlpp::TextNode* textnode=dynamic_cast<const xmlpp::TextNode*>(src);
 	if(textnode)
 	{
 		return buildFromString(textnode->get_content(), false);
 	}
+	else if ((contentnode = dynamic_cast<const xmlpp::ContentNode*>(src)))
+	{
+		// ContentNode but not TextNode => comment, PI or CData
+		// These can't be root nodes so we add a dummy root.
+		// TODO: The root node should not be accessible from
+		// AS code.
+		xmlpp::Element* root = parser.get_document()->create_root_node("dummy_root");
+		return root->import_node(contentnode);
+	}
 	else
 	{
 		return parser.get_document()->create_root_node_by_import(src);
-- 
1.8.5.3

