From 7402cf9e4b7bb8714db79a4b13f96febd2b23fcf Mon Sep 17 00:00:00 2001
From: Antti Ajanki <antti.ajanki@iki.fi>
Date: Tue, 18 Jun 2013 20:34:44 +0300
Subject: [PATCH 071/110] [opcodes] Throw correct exception in checkfilter

---
 src/scripting/abc_opcodes.cpp | 9 +++------
 1 file changed, 3 insertions(+), 6 deletions(-)

diff --git a/src/scripting/abc_opcodes.cpp b/src/scripting/abc_opcodes.cpp
index 2d779da..dd1fb45 100644
--- a/src/scripting/abc_opcodes.cpp
+++ b/src/scripting/abc_opcodes.cpp
@@ -140,12 +140,9 @@ void ABCVm::coerce_a()
 
 ASObject* ABCVm::checkfilter(ASObject* o)
 {
-	Class_base* xmlClass=Class<XML>::getClass();
-	Class_base* xmlListClass=Class<XMLList>::getClass();
-
-	if (o->getClass()!=xmlClass && o->getClass()!=xmlListClass)
-		throw Class<TypeError>::getInstanceS();
-
+	LOG(LOG_CALLS, _("checkfilter") );
+	if (!o->is<XML>() && !o->is<XMLList>())
+		throwError<TypeError>(kFilterError, o->getClassName());
 	return o;
 }
 
-- 
1.8.5.3

