From f48d0357a8a4992610a1008a27888d62ec70abd2 Mon Sep 17 00:00:00 2001
From: Antti Ajanki <antti.ajanki@iki.fi>
Date: Tue, 18 Jun 2013 22:23:38 +0300
Subject: [PATCH 072/110] [opcodes] null or undefined can not be constructed

---
 src/scripting/abc_opcodes.cpp | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/src/scripting/abc_opcodes.cpp b/src/scripting/abc_opcodes.cpp
index dd1fb45..3b96439 100644
--- a/src/scripting/abc_opcodes.cpp
+++ b/src/scripting/abc_opcodes.cpp
@@ -654,15 +654,6 @@ void ABCVm::construct(call_context* th, int m)
 			break;
 		}
 
-		case T_UNDEFINED:
-		case T_NULL:
-		{
-			//Inc ref count to make up for decremnt later
-			obj->incRef();
-			ret=obj;
-			break;
-		}
-
 		case T_FUNCTION:
 		{
 			ret = constructFunction(th, obj->as<IFunction>(), args, m);
-- 
1.8.5.3

