From 458614be9440b5c664204a94feaee8c2ce1937ca Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ludger=20Kr=C3=A4mer?= <dbluelle@blau-weissoedingen.de>
Date: Sun, 18 Aug 2013 13:03:02 +0200
Subject: [PATCH 085/110] ensure that valueOf (if available) is used before
 using toInt

---
 src/asobject.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/asobject.cpp b/src/asobject.cpp
index 2c3383c..94fab45 100644
--- a/src/asobject.cpp
+++ b/src/asobject.cpp
@@ -227,7 +227,7 @@ uint16_t ASObject::toUInt16()
 
 int32_t ASObject::toInt()
 {
-	return 0;
+	return toPrimitive()->toInt();
 }
 
 /* Implements ECMA's ToPrimitive (9.1) and [[DefaultValue]] (8.6.2.6) */
-- 
1.8.5.3

