From 8a74f28a5a611a6307f59334c626e1ed43fe876c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ludger=20Kr=C3=A4mer?= <dbluelle@blau-weissoedingen.de>
Date: Sat, 24 Aug 2013 12:10:47 +0200
Subject: [PATCH 088/110] remove wrong assert

---
 src/scripting/flash/geom/flashgeom.cpp | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/scripting/flash/geom/flashgeom.cpp b/src/scripting/flash/geom/flashgeom.cpp
index 9cd0c84..aa3b328 100644
--- a/src/scripting/flash/geom/flashgeom.cpp
+++ b/src/scripting/flash/geom/flashgeom.cpp
@@ -582,7 +582,6 @@ ASFUNCTIONBODY(ColorTransform,setColor)
 
 ASFUNCTIONBODY(ColorTransform,getColor)
 {
-	assert_and_throw(argslen==1);
 	ColorTransform* th=static_cast<ColorTransform*>(obj);
 
 	int ao, ro, go, bo;
-- 
1.8.5.3

