From 1fcc99310547ba02aa844d5aa27da65b2eca3f70 Mon Sep 17 00:00:00 2001
From: Fabian Ebner <f.ebner94@gmail.com>
Date: Fri, 13 Sep 2013 21:37:08 +0200
Subject: [PATCH 094/110] Count Tamarin tests which passed all test cases

---
 tests/tests | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/tests/tests b/tests/tests
index e3934e7..e89bbc5 100755
--- a/tests/tests
+++ b/tests/tests
@@ -368,6 +368,9 @@ for test in $TESTS; do
 		FAILURECOUNT=`expr $FAILURECOUNT + $THISFAILEDCOUNT`;
 		SUCCESSCOUNT=`expr $SUCCESSCOUNT + $THISSUCCESSCOUNT`;
 		FAILED_TESTS="$FAILED_TESTS $test"
+		if [ $THISFAILEDCOUNT -eq 0 ]; then
+			ALLSUCCESSCOUNT=`expr $ALLSUCCESSCOUNT + 1`
+		fi
 	else
 		if [ $COLORS -eq 1 ]; then
 			echo -e "\\e[0;31m$test printed no test summary!\\e[0m"
-- 
1.8.5.3

