From 4d2d9279e931063df24690f18f7635e47a958141 Mon Sep 17 00:00:00 2001
From: Antti Ajanki <antti.ajanki@iki.fi>
Date: Sat, 14 Sep 2013 16:43:38 +0300
Subject: [PATCH 095/110] Add an include file (needed on Debian unstable)

---
 src/backends/decoder.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/backends/decoder.h b/src/backends/decoder.h
index 28bd232..35dbaab 100644
--- a/src/backends/decoder.h
+++ b/src/backends/decoder.h
@@ -30,6 +30,7 @@ extern "C"
 #include <libavformat/avformat.h>
 #include <libavresample/avresample.h>
 #include <libavutil/opt.h>
+#include <libavutil/mathematics.h>
 #ifndef AVCODEC_MAX_AUDIO_FRAME_SIZE
 #define AVCODEC_MAX_AUDIO_FRAME_SIZE 192000 // 1 second of 48khz 32bit audio
 #endif
-- 
1.8.5.3

