From 50c8bc8f171594a15c5250b071bf695712d33557 Mon Sep 17 00:00:00 2001
From: Antti Ajanki <antti.ajanki@iki.fi>
Date: Sat, 9 Nov 2013 19:23:31 +0200
Subject: [PATCH 106/110] Override sandbox only when handling a remote --url

---
 src/main.cpp | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/main.cpp b/src/main.cpp
index d3ef2d9..6f8d331 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -287,7 +287,12 @@ int main(int argc, char* argv[])
 	{
 		sys->mainClip->setOrigin(url, fileName);
 		sys->parseParametersFromURL(sys->mainClip->getOrigin());
-		sandboxType = SecurityManager::REMOTE;
+		if (sandboxType != SecurityManager::REMOTE &&
+		    sys->mainClip->getOrigin().getProtocol() != "file")
+		{
+			LOG(LOG_INFO, _("Switching to remote sandbox because of remote url"));
+			sandboxType = SecurityManager::REMOTE;
+		}
 	}
 #ifndef _WIN32
 	//When running in a local sandbox, set the root URL to the current working dir
-- 
1.8.5.3

