From 7c1e525d5c19fc778d40b8ec34aa9d3e17d71c19 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ludger=20Kr=C3=A4mer?= <dbluelle@blau-weissoedingen.de>
Date: Sun, 17 Nov 2013 10:07:35 +0100
Subject: [PATCH 107/110] [ASObject] handle propertyIsEnumerable without
 parameter

---
 src/asobject.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/asobject.cpp b/src/asobject.cpp
index ce9d6ff..3c890fc 100644
--- a/src/asobject.cpp
+++ b/src/asobject.cpp
@@ -942,7 +942,8 @@ ASFUNCTIONBODY(ASObject,isPrototypeOf)
 
 ASFUNCTIONBODY(ASObject,propertyIsEnumerable)
 {
-	assert_and_throw(argslen==1);
+	if (argslen == 0)
+		return abstract_b(false);
 	multiname name(NULL);
 	name.name_type=multiname::NAME_STRING;
 	name.name_s_id=getSys()->getUniqueStringId(args[0]->toString());
-- 
1.8.5.3

