From 631c84c0b9c1ed361de390f07d6f062ea19bd9ce Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ludger=20Kr=C3=A4mer?= <dbluelle@blau-weissoedingen.de>
Date: Sun, 17 Nov 2013 10:09:17 +0100
Subject: [PATCH 108/110] [ABCContext] fix multiname handling

---
 src/scripting/abc.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/scripting/abc.cpp b/src/scripting/abc.cpp
index 19343b0..59b2714 100644
--- a/src/scripting/abc.cpp
+++ b/src/scripting/abc.cpp
@@ -782,6 +782,11 @@ multiname* ABCContext::getMultinameImpl(ASObject* n, ASObject* n2, unsigned int
 				ret->ns.clear();
 				ret->ns.push_back(nsNameAndKind(qname->getURI(),NAMESPACE));
 			}
+			else
+			{
+				ret->ns.clear();
+				ret->ns.push_back(nsNameAndKind("",NAMESPACE));
+			}
 
 			ret->setName(n);
 			n->decRef();
-- 
1.8.5.3

