--- src/Font.py
+++ src/Font.py
@@ -206,7 +206,7 @@ class Font:
 
       # Draw outlines
       """
-      import Image, ImageFilter
+      from PIL import Image, ImageFilter
       srcImg = Image.fromstring("RGBA", s.get_size(), pygame.image.tostring(s, "RGBA"))
       img    = Image.fromstring("RGBA", s.get_size(), pygame.image.tostring(s, "RGBA"))
       for y in xrange(img.size[1]):
--- src/Texture.py
+++ src/Texture.py
@@ -24,10 +24,10 @@ from __future__ import division
 
 import Log
 import Config
-import Image
+from PIL import Image
 import pygame
 import StringIO
-import PngImagePlugin
+from PIL import PngImagePlugin
 from OpenGL.GL import *
 from OpenGL.GLU import *
 from Queue import Queue, Empty
