--- softdevice-cvs-20090630.orig/VideoFilter.h
+++ softdevice-cvs-20090630/VideoFilter.h
@@ -109,8 +109,8 @@ public:
 class cLibAvPostProc : public cVideoFilter {
         int width, height;
         PixelFormat pix_fmt;
-        pp_mode_t *ppmode;
-        pp_context_t *ppcontext;
+        pp_mode *ppmode;
+        pp_context *ppcontext;
         sPicBuffer *outBuf;
         int currentDeintMethod,currentppMethod,currentppQuality;
 public:

