diff -ruN ZoneMinder-1.25.0.orig/web/Makefile.am ZoneMinder-1.25.0/web/Makefile.am
--- ZoneMinder-1.25.0.orig/web/Makefile.am	2011-05-15 23:15:23.000000000 +0200
+++ ZoneMinder-1.25.0/web/Makefile.am	2011-09-18 00:24:00.428778534 +0200
@@ -22,13 +22,12 @@
 
 # Yes, you are correct. This is a HACK!
 install-data-hook:
-	( cd $(DESTDIR)$(webdir); chown $(webuser):$(webgroup) $(dist_web_DATA) )
-	( cd $(DESTDIR)$(webdir); chown -R $(webuser):$(webgroup) $(SUBDIRS) )
-	@-( cd $(DESTDIR)$(webdir); if ! test -e events; then mkdir events; fi; chown $(webuser):$(webgroup) events; chmod u+w events )
-	@-( cd $(DESTDIR)$(webdir); if ! test -e images; then mkdir images; fi; chown $(webuser):$(webgroup) images; chmod u+w images )
-	@-( cd $(DESTDIR)$(webdir); if ! test -e sounds; then mkdir sounds; fi; chown $(webuser):$(webgroup) sounds; chmod u+w sounds )
-	@-( cd $(DESTDIR)$(webdir); if ! test -e tools; then mkdir tools; fi; chown $(webuser):$(webgroup) tools; chmod u+w tools )
-	@-( cd $(DESTDIR)$(webdir); if ! test -e temp; then mkdir temp; fi; chown $(webuser):$(webgroup) temp; chmod u+w temp )
+	( cd $(DESTDIR)$(webdir) )
+	@-( cd $(DESTDIR)$(webdir); if ! test -e events; then mkdir events; fi )
+	@-( cd $(DESTDIR)$(webdir); if ! test -e images; then mkdir images; fi )
+	@-( cd $(DESTDIR)$(webdir); if ! test -e sounds; then mkdir sounds; fi )
+	@-( cd $(DESTDIR)$(webdir); if ! test -e tools; then mkdir tools; fi )
+	@-( cd $(DESTDIR)$(webdir); if ! test -e temp; then mkdir temp; fi )
 
 uninstall-hook:
 	@-( cd $(DESTDIR)$(webdir); rm -rf $(SUBDIRS) )
