Index: blender-2.47/source/blender/src/hddaudio.c
===================================================================
--- blender-2.47.orig/source/blender/src/hddaudio.c
+++ blender-2.47/source/blender/src/hddaudio.c
@@ -302,7 +302,8 @@ static void sound_hdaudio_extract_small_
 				audio_pkt_size = packet.size;
 
 				while (audio_pkt_size > 0) {
-					len = avcodec_decode_audio(
+					data_size= AVCODEC_MAX_AUDIO_FRAME_SIZE;
+					len = avcodec_decode_audio2(
 						hdaudio->pCodecCtx, 
 						hdaudio->decode_cache 
 						+ decode_pos, 
@@ -469,7 +470,8 @@ static void sound_hdaudio_extract_small_
 			}
 
 			while (audio_pkt_size > 0) {
-				len = avcodec_decode_audio(
+				data_size = AVCODEC_MAX_AUDIO_FRAME_SIZE;
+				len = avcodec_decode_audio2(
 					hdaudio->pCodecCtx, 
 					hdaudio->decode_cache 
 					+ decode_pos, 

