--- blender-2.49b.orig//extern/bullet2/src/BulletSoftBody/btSoftBodyInternals.h
+++ blender-2.49b/extern/bullet2/src/BulletSoftBody/btSoftBodyInternals.h
@@ -25,6 +25,8 @@ subject to the following restrictions:
 #include "BulletCollision/CollisionShapes/btConvexInternalShape.h"
 #include "BulletCollision/NarrowPhaseCollision/btGjkEpa2.h"
 
+#include <string.h>
+
 //
 // btSymMatrix
 //
@@ -171,8 +173,7 @@ public:
 template <typename T>
 static inline void			ZeroInitialize(T& value)
 {
-	static const T	zerodummy;
-	value=zerodummy;
+	memset(&value,0,sizeof(T));
 }
 //
 template <typename T>
--- blender-2.49b.orig//intern/memutil/MEM_Allocator.h
+++ blender-2.49b/intern/memutil/MEM_Allocator.h
@@ -24,6 +24,8 @@
 #ifndef __MEM_Allocator_h_included__
 #define __MEM_Allocator_h_included__ 1
 
+#include <cstddef>
+
 #include "guardedalloc/MEM_guardedalloc.h"
 #include "guardedalloc/BLO_sys_types.h"
 
--- blender-2.49b.orig//source/gameengine/Physics/Dummy/DummyPhysicsEnvironment.h
+++ blender-2.49b/source/gameengine/Physics/Dummy/DummyPhysicsEnvironment.h
@@ -29,6 +29,8 @@
 #ifndef _DUMMYPHYSICSENVIRONMENT
 #define _DUMMYPHYSICSENVIRONMENT
 
+#include <cstdlib>
+
 #include "PHY_IPhysicsEnvironment.h"
 
 /**
--- blender-2.49b.orig/source/gameengine/Converter/BL_ActionActuator.h
+++ blender-2.49b/source/gameengine/Converter/BL_ActionActuator.h
@@ -30,6 +30,8 @@
 #ifndef BL_ACTIONACTUATOR
 #define BL_ACTIONACTUATOR
 
+#include <cstddef>
+
 #include "GEN_HashedPtr.h"
 #include "SCA_IActuator.h"
 #include "DNA_actuator_types.h"
--- blender-2.49b.orig/source/gameengine/GameLogic/SCA_ILogicBrick.h
+++ blender-2.49b/source/gameengine/GameLogic/SCA_ILogicBrick.h
@@ -34,6 +34,7 @@
 #include "BoolValue.h"
 #include "GEN_Map.h"
 #include "GEN_HashedPtr.h"
+#include <cstddef>
 
 class SCA_ILogicBrick : public CValue
 {
--- blender-2.49b.orig/source/gameengine/GameLogic/SCA_ISensor.h
+++ blender-2.49b/source/gameengine/GameLogic/SCA_ISensor.h
@@ -35,6 +35,7 @@
 #include "SCA_IController.h"
 
 #include <vector>
+#include <cstddef>
 
 /**
  * Interface Class for all logic Sensors. Implements
--- blender-2.49b.orig/source/gameengine/GameLogic/SCA_KeyboardSensor.h
+++ blender-2.49b/source/gameengine/GameLogic/SCA_KeyboardSensor.h
@@ -34,6 +34,7 @@
 #include "SCA_ISensor.h"
 #include "BoolValue.h"
 #include <list>
+#include <cstddef>
 
 /**
  * The keyboard sensor listens to the keyboard, and passes on events
--- blender-2.49b.orig/source/gameengine/GameLogic/SCA_RandomSensor.h
+++ blender-2.49b/source/gameengine/GameLogic/SCA_RandomSensor.h
@@ -35,6 +35,7 @@
 #include "SCA_ISensor.h"
 #include "BoolValue.h"
 #include "SCA_RandomNumberGenerator.h"
+#include <cstddef>
 
 class SCA_RandomSensor : public SCA_ISensor
 {
--- blender-2.49b.orig/source/gameengine/Rasterizer/RAS_IPolygonMaterial.h
+++ blender-2.49b/source/gameengine/Rasterizer/RAS_IPolygonMaterial.h
@@ -29,6 +29,8 @@
 #ifndef __RAS_IPOLYGONMATERIAL
 #define __RAS_IPOLYGONMATERIAL
 
+#include <cstddef>
+
 #include "STR_HashedString.h"
 
 /**

