Index: source/blender/render/intern/source/rendercore.c
===================================================================
--- source/blender/render/intern/source/rendercore.c	(revision 21781)
+++ source/blender/render/intern/source/rendercore.c	(working copy)
@@ -2241,7 +2241,7 @@
 	/* no direction checking for now, doesn't always improve the result
 	 * (INPR(shi->facenor, bs->dir) > 0.0f); */
 
-	return (R.objectinstance[ob].obr->ob != bs->actob);
+	return (R.objectinstance[ob & ~RE_RAY_TRANSFORM_OFFS].obr->ob != bs->actob);
 }
 
 static int bake_intersect_tree(RayTree* raytree, Isect* isect, float *start, float *dir, float sign, float *hitco, float *dist)
