--- open-vm-tools-2011.11.20-535097.orig/modules/linux/vmhgfs/tcp.c
+++ open-vm-tools-2011.11.20-535097/modules/linux/vmhgfs/tcp.c
@@ -28,6 +28,7 @@
 /* Must come before any kernel header file. */
 #include "driver-config.h"
 
+#include <linux/module.h>
 #include <linux/socket.h>
 #include <linux/in.h>
 #include <linux/net.h>

--- open-vm-tools-2011.11.20-535097.orig/modules/linux/vmhgfs/fsutil.c
+++ open-vm-tools-2011.11.20-535097/modules/linux/vmhgfs/fsutil.c
@@ -607,7 +607,7 @@ HgfsChangeFileAttributes(struct inode *i
     * account for '.' and ".."), and find printed a hard link error. So until
     * we have getattr support for nlink, everyone gets 1.
     */
-   inode->i_nlink = 1;
+   set_nlink(inode, 1);
 
    /*
     * Use the stored uid and gid if we were given them at mount-time, or if

