--- open-vm-tools-2012.12.26-958366.orig/modules/linux/vmhgfs/inode.c
+++ open-vm-tools-2012.12.26-958366/modules/linux/vmhgfs/inode.c
@@ -1811,14 +1811,13 @@ HgfsPermission(struct inode *inode,
     */
    if (mask & MAY_ACCESS) { /* For sys_access. */
       struct dentry *dentry;
-      struct hlist_node *p;
       int dcount = 0;
 
       if (mask & MAY_NOT_BLOCK)
          return -ECHILD;
 
       /* Find a dentry with valid d_count. Refer bug 587879. */
-      hlist_for_each_entry(dentry, p, &inode->i_dentry, d_alias) {
+      hlist_for_each_entry(dentry, &inode->i_dentry, d_alias) {
          dcount = dentry->d_count;
          if (dcount) {
             LOG(4, ("Found %s %d \n", dentry->d_name.name, dcount));

