--- lxqt-config-monitor/~monitor.h	2015-11-11 14:25:38.684040789 +0100
+++ lxqt-config-monitor/monitor.h	2015-11-11 14:25:52.895761815 +0100
@@ -20,6 +20,7 @@
 #ifndef _MONITOR_H_
 #define _MONITOR_H_
 
+#include <QObject>
 #include <QStringList>
 #include <QHash>
 #include <QList>
@@ -79,4 +80,4 @@
 /**Gets size from string rate. String rate format is "widthxheight". Example: 800x600*/
 QSize sizeFromString(QString str);
 
-#endif // _MONITOR_H_
\ Manca newline alla fine del file
+#endif // _MONITOR_H_
