From 46dd97e115ddc7219c88b0818c4d5b235162fe6e Mon Sep 17 00:00:00 2001
From: Michal Kottman <michal.kottman@gmail.com>
Date: Sun, 23 Jun 2013 22:23:35 +0200
Subject: [PATCH 5/6] Add call for AMD HD8870, fix #37

---
 examples/turn_off_gpu.sh | 1 +
 1 file changed, 1 insertion(+)

diff --git a/examples/turn_off_gpu.sh b/examples/turn_off_gpu.sh
index 14db28f..37e26a0 100755
--- a/examples/turn_off_gpu.sh
+++ b/examples/turn_off_gpu.sh
@@ -29,6 +29,7 @@ methods="
 \_SB.PCI0.P0P2.VGA.PX02
 \_SB_.PCI0.PEGP.DGFX._OFF
 \_SB_.PCI0.VGA.PX02
+\_SB.PCI0.PEG0.PEGP.SGOF
 "
 
 for m in $methods; do
-- 
1.8.2.1

