--- grass-6.4.1.orig/lib/ogsf/gsd_img_mpeg.c
+++ grass-6.4.1/lib/ogsf/gsd_img_mpeg.c
@@ -66,7 +66,7 @@ static AVStream *add_video_stream(AVForm
 
     c = st->codec;
     c->codec_id = codec_id;
-    c->codec_type = CODEC_TYPE_VIDEO;
+    c->codec_type = AVMEDIA_TYPE_VIDEO;
 
     /* put sample parameters */
     c->bit_rate = 400000;
@@ -215,7 +215,7 @@ static void write_video_frame(AVFormatCo
 
 	av_init_packet(&pkt);
 
-	pkt.flags |= PKT_FLAG_KEY;
+	pkt.flags |= AV_PKT_FLAG_KEY;
 	pkt.stream_index = st->index;
 	pkt.data = (uint8_t *) picture;
 	pkt.size = sizeof(AVPicture);
@@ -236,7 +236,7 @@ static void write_video_frame(AVFormatCo
 		av_rescale_q(c->coded_frame->pts, c->time_base,
 			     st->time_base);
 	    if (c->coded_frame->key_frame)
-		pkt.flags |= PKT_FLAG_KEY;
+		pkt.flags |= AV_PKT_FLAG_KEY;
 	    pkt.stream_index = st->index;
 	    pkt.data = video_outbuf;
 	    pkt.size = out_size;

