--- open-vm-tools-2011.10.26-514583.orig/services/plugins/dndcp/dndGuest/copyPasteRpcV3.cc	2011-10-27 18:57:11.000000000 +0000
+++ open-vm-tools-2011.10.26-514583/services/plugins/dndcp/dndGuest/copyPasteRpcV3.cc	2011-11-14 19:11:43.947481617 +0000
@@ -22,7 +22,7 @@
  *     Implementation of the CopyPasteRpcV3 interface.
  */
 
-
+#include <cstddef>
 #include "copyPasteRpcV3.hh"
 
 extern "C" {
--- open-vm-tools-2011.10.26-514583.orig/services/plugins/dndcp/dndGuest/copyPasteRpcV4.cc	2011-10-27 18:57:11.000000000 +0000
+++ open-vm-tools-2011.10.26-514583/services/plugins/dndcp/dndGuest/copyPasteRpcV4.cc	2011-11-14 19:15:13.692942784 +0000
@@ -22,6 +22,7 @@
  * Rpc layer object for CopyPaste version 4.
  */
 
+#include <cstddef>
 #include "copyPasteRpcV4.hh"
 
 extern "C" {
--- open-vm-tools-2011.10.26-514583.orig/services/plugins/dndcp/dndGuest/dndRpcV3.cc	2011-10-27 18:57:11.000000000 +0000
+++ open-vm-tools-2011.10.26-514583/services/plugins/dndcp/dndGuest/dndRpcV3.cc	2011-11-14 19:12:33.810689395 +0000
@@ -35,6 +35,8 @@
 #include <afx.h>
 #endif
 
+#include <cstddef>
+
 #include "dndRpcV3.hh"
 
 extern "C" {
--- open-vm-tools-2011.10.26-514583.orig/services/plugins/dndcp/dndGuest/dndRpcV4.cc	2011-10-27 18:57:11.000000000 +0000
+++ open-vm-tools-2011.10.26-514583/services/plugins/dndcp/dndGuest/dndRpcV4.cc	2011-11-14 19:15:27.630441831 +0000
@@ -22,7 +22,7 @@
  * Implementation of the DnDRpcV4 object.
  */
 
-
+#include <cstddef>
 #include "dndRpcV4.hh"
 
 extern "C" {
--- open-vm-tools-2011.10.26-514583.orig/services/plugins/dndcp/dndGuest/fileTransferRpcV4.cc	2011-10-27 18:57:11.000000000 +0000
+++ open-vm-tools-2011.10.26-514583/services/plugins/dndcp/dndGuest/fileTransferRpcV4.cc	2011-11-14 19:15:44.551833630 +0000
@@ -22,6 +22,7 @@
  * Implementation of common layer file transfer rpc version 4 object.
  */
 
+#include <cstddef>
 #include "fileTransferRpcV4.hh"
 #include "dndCPTransport.h"
 
--- open-vm-tools-2011.10.26-514583.orig/services/plugins/dndcp/dndGuest/guestCopyPasteDest.cc	2011-10-27 18:57:11.000000000 +0000
+++ open-vm-tools-2011.10.26-514583/services/plugins/dndcp/dndGuest/guestCopyPasteDest.cc	2011-11-14 19:12:55.959893292 +0000
@@ -22,6 +22,7 @@
  * Implementation of common layer GuestCopyPasteDest object for guest.
  */
 
+#include <cstddef>
 #include "guestCopyPaste.hh"
 
 extern "C" {
--- open-vm-tools-2011.10.26-514583.orig/services/plugins/dndcp/dndGuest/guestCopyPasteMgr.cc	2011-10-27 18:57:11.000000000 +0000
+++ open-vm-tools-2011.10.26-514583/services/plugins/dndcp/dndGuest/guestCopyPasteMgr.cc	2011-11-14 19:13:09.145419367 +0000
@@ -22,6 +22,7 @@
  * Implementation of common layer GuestCopyPasteMgr object for guest.
  */
 
+#include <cstddef>
 #include "guestCopyPaste.hh"
 #include "copyPasteRpcV3.hh"
 #include "copyPasteRpcV4.hh"
--- open-vm-tools-2011.10.26-514583.orig/services/plugins/dndcp/dndGuest/guestCopyPasteSrc.cc	2011-10-27 18:57:11.000000000 +0000
+++ open-vm-tools-2011.10.26-514583/services/plugins/dndcp/dndGuest/guestCopyPasteSrc.cc	2011-11-14 19:13:22.330945443 +0000
@@ -22,6 +22,7 @@
  * Implementation of common layer GuestCopyPasteSrc object for guest.
  */
 
+#include <cstddef>
 #include "guestCopyPaste.hh"
 
 extern "C" {
--- open-vm-tools-2011.10.26-514583.orig/services/plugins/dndcp/dndGuest/guestDnDCPMgr.cc	2011-10-27 18:57:11.000000000 +0000
+++ open-vm-tools-2011.10.26-514583/services/plugins/dndcp/dndGuest/guestDnDCPMgr.cc	2011-11-14 19:13:40.901277974 +0000
@@ -22,6 +22,7 @@
  * Implementation of common layer GuestDnDCPMgr object.
  */
 
+#include <cstddef>
 #include "guestDnDCPMgr.hh"
 #ifdef DND_USING_VMCF
 #   include "dndCPTransportVMCF.hh"
--- open-vm-tools-2011.10.26-514583.orig/services/plugins/dndcp/dndGuest/guestDnDDest.cc	2011-10-27 18:57:11.000000000 +0000
+++ open-vm-tools-2011.10.26-514583/services/plugins/dndcp/dndGuest/guestDnDDest.cc	2011-11-14 19:13:56.021734503 +0000
@@ -22,7 +22,7 @@
  * Implementation of common layer GuestDnDDest object for guest.
  */
 
-
+#include <cstddef>
 #include "guestDnD.hh"
 
 extern "C" {
--- open-vm-tools-2011.10.26-514583.orig/services/plugins/dndcp/dndGuest/guestDnDMgr.cc	2011-10-27 18:57:11.000000000 +0000
+++ open-vm-tools-2011.10.26-514583/services/plugins/dndcp/dndGuest/guestDnDMgr.cc	2011-11-14 19:14:08.095300545 +0000
@@ -22,6 +22,7 @@
  * Implementation of common layer GuestDnDMgr object for guest.
  */
 
+#include <cstddef>
 #include "guestDnD.hh"
 #include "dndRpcV4.hh"
 #include "dndRpcV3.hh"
--- open-vm-tools-2011.10.26-514583.orig/services/plugins/dndcp/dndGuest/guestDnDSrc.cc	2011-10-27 18:57:11.000000000 +0000
+++ open-vm-tools-2011.10.26-514583/services/plugins/dndcp/dndGuest/guestDnDSrc.cc	2011-11-14 19:14:39.894157608 +0000
@@ -22,6 +22,7 @@
  * Implementation of common layer GuestDnDSrc object for guest.
  */
 
+#include <cstddef>
 #include "guestDnD.hh"
 
 extern "C" {
--- open-vm-tools-2011.10.26-514583.orig/services/plugins/dndcp/dndGuest/guestFileTransfer.cc	2011-10-27 18:57:11.000000000 +0000
+++ open-vm-tools-2011.10.26-514583/services/plugins/dndcp/dndGuest/guestFileTransfer.cc	2011-11-14 19:14:59.579450062 +0000
@@ -22,6 +22,7 @@
  * Implementation of common layer file transfer object for guest.
  */
 
+#include <cstddef>
 #include "guestFileTransfer.hh"
 #include "fileTransferRpcV4.hh"
 
