--- SConstruct.orig	2010-07-25 01:45:25.000000000 +0200
+++ SConstruct	2010-07-25 01:45:50.000000000 +0200
@@ -300,7 +300,6 @@
   'gcc': {
     'all': {
       'WARNINGFLAGS': ['-Wall',
-                       '-Werror',
                        '-W',
                        '-Wno-unused-parameter',
                        '-Wnon-virtual-dtor']
@@ -381,7 +380,7 @@
 DTOA_EXTRA_FLAGS = {
   'gcc': {
     'all': {
-      'WARNINGFLAGS': ['-Werror', '-Wno-uninitialized'],
+      'WARNINGFLAGS': ['-Wno-uninitialized'],
       'CCFLAGS': GCC_DTOA_EXTRA_CCFLAGS
     }
   },
