--- build/premake/premake4/src/actions/make/make_cpp.lua.orig	2011-12-17 20:59:29.000000000 +0800
+++ build/premake/premake4/src/actions/make/make_cpp.lua	2011-12-25 07:27:40.268759001 +0800
@@ -248,7 +248,7 @@
 		-- set up precompiled headers
 		_.pchconfig(cfg)
 				
-		_p('  CFLAGS    += $(CPPFLAGS) $(ARCH) %s', table.concat(table.join(cc.getcflags(cfg), cfg.buildoptions), " "))
+		_p('  CFLAGS    += $(CPPFLAGS) %s', table.concat(table.join(cc.getcflags(cfg), cfg.buildoptions), " "))
 		_p('  CXXFLAGS  += $(CFLAGS) %s', table.concat(cc.getcxxflags(cfg), " "))
 		_p('  LDFLAGS   += %s', table.concat(table.join(cc.getldflags(cfg), cfg.linkoptions, cc.getlibdirflags(cfg)), " "))
 		_p('  LIBS      += %s %s', table.concat(cc.getlinkflags(cfg), " "), table.concat(cfg.gnuexternals, " "))
@@ -272,7 +272,7 @@
 			else
 				lddeps = '-Xlinker --start-group $(LDDEPS) -Xlinker --end-group'
 			end
-			_p('  LINKCMD    = $(%s) -o $(TARGET) $(OBJECTS) $(LDFLAGS) $(RESOURCES) $(ARCH) %s $(LIBS)', 
+			_p('  LINKCMD    = $(%s) -o $(TARGET) $(OBJECTS) $(LDFLAGS) $(RESOURCES) %s $(LIBS)', 
 			iif(cfg.language == "C", "CC", "CXX"), lddeps)
 		end
 		
--- build/workspaces/update-workspaces.sh.orig	2011-11-15 00:10:41.000000000 +0800
+++ build/workspaces/update-workspaces.sh	2011-12-25 07:30:39.729759000 +0800
@@ -62,6 +62,8 @@
 # Now build premake and run it to create the makefiles
 cd ../premake/premake4
 make -C build/gmake.unix ${JOBS} || die "Premake build failed"
+./bin/release/premake4 embed
+make -C build/gmake.unix ${JOBS} || die "Premake build failed"
 
 echo
 
