commit 73ae7f3ae54af48712f2faf3c4809c0c225d6692
Author: Fabio Erculiani <lxnay@sabayon.org>
Date:   Sun Feb 24 13:42:37 2013 +0000

    entropy: add compatibility with Entropy 179

diff --git a/backends/entropy/entropyBackend.py b/backends/entropy/entropyBackend.py
index 0c3c29b..eb2c50d 100755
--- a/backends/entropy/entropyBackend.py
+++ b/backends/entropy/entropyBackend.py
@@ -1336,7 +1336,14 @@ class PackageKitEntropyBackend(PackageKitBaseBackend, PackageKitEntropyMixin):
         # this is the part that takes time
         self.percentage(0)
         try:
-            update, remove, fine, spm_fine = self._entropy.calculate_updates()
+            outcome = self._entropy.calculate_updates()
+
+            if isinstance(outcome, dict):
+                updates, remove, fine, spm_fine = outcome['update'], \
+                    outcome['remove'], outcome['fine'], outcome['spm_fine']
+            else:
+                updates, remove, fine, spm_fine = outcome
+
         except SystemDatabaseError as err:
             self.error(ERROR_DEP_RESOLUTION_FAILED,
                 "System Repository error: %s" % (err,))
