From 9cb4fe58aa212de8067b3df585c425efbb736274 Mon Sep 17 00:00:00 2001
From: Fabio Erculiani <lxnay@sabayon.org>
Date: Sun, 24 Nov 2013 16:04:51 +0100
Subject: [PATCH] entropy: fix package_path() usage

---
 backends/entropy/entropyBackend.py | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/backends/entropy/entropyBackend.py b/backends/entropy/entropyBackend.py
index 6a53ee2..57b6c59 100755
--- a/backends/entropy/entropyBackend.py
+++ b/backends/entropy/entropyBackend.py
@@ -688,12 +688,14 @@ class PackageKitEntropyMixin(object):
                 package = self._entropy.Package()
                 package.prepare(match, self._fetch_action, metaopts)
                 x_rc = package.run()
+                package_path = package.pkgmeta['pkgpath']
                 package.kill()
             else:
                 package = self._action_factory.get(
                     self._fetch_action, match,
                     opts = metaopts)
                 x_rc = package.start()
+                package_path = package.package_path()
                 package.finalize()
 
             if x_rc != 0:
@@ -703,9 +705,9 @@ class PackageKitEntropyMixin(object):
 
             # emit the file we downloaded
             if self._action_factory is None:
-                self.files(pk_pkg, package.pkgmeta['pkgpath'])
+                self.files(pk_pkg, package_path)
             else:
-                self.files(pk_pkg, package.package_path())
+                self.files(pk_pkg, package_path)
 
         # spawn UGC
         if not simulate:
-- 
1.8.4.3

