diff --git a/backends/entropy/entropyBackend.py b/backends/entropy/entropyBackend.py
index 7854725..0c3c29b 100755
--- a/backends/entropy/entropyBackend.py
+++ b/backends/entropy/entropyBackend.py
@@ -62,6 +62,7 @@ from entropy.i18n import _, _LOCALE
 from entropy.const import etpConst, const_convert_to_rawstring, \
     const_convert_to_unicode, const_get_stringtype
 from entropy.client.interfaces import Client
+from entropy.client.interfaces.db import InstalledPackagesRepository
 from entropy.core.settings.base import SystemSettings
 from entropy.misc import LogFile
 from entropy.cache import EntropyCacher
@@ -156,7 +157,10 @@ class PackageKitEntropyMixin(object):
                 "Invalid metadata passed")
 
         # if installed, repo should be 'installed', packagekit rule
-        if repo_name == etpConst['clientdbid']:
+        cl_repo_name = etpConst.get(
+            'clientdbid', # forward compatibility
+            getattr(InstalledPackagesRepository, "NAME", None))
+        if repo_name == cl_repo_name:
             repo_name = "installed"
 
         # openoffice-clipart;2.6.22;ppc64;fedora
