commit 02d59f97bff92b87117a4dddc705179500f16f52
Author: Fabio Erculiani <lxnay@sabayon.org>
Date:   Tue Mar 19 22:53:07 2013 +0000

    entropy: fix undefined reference "update"

diff --git a/backends/entropy/entropyBackend.py b/backends/entropy/entropyBackend.py
index eb2c50d..9ae59e1 100755
--- a/backends/entropy/entropyBackend.py
+++ b/backends/entropy/entropyBackend.py
@@ -1339,10 +1339,10 @@ class PackageKitEntropyBackend(PackageKitBaseBackend, PackageKitEntropyMixin):
             outcome = self._entropy.calculate_updates()
 
             if isinstance(outcome, dict):
-                updates, remove, fine, spm_fine = outcome['update'], \
+                update, remove, fine, spm_fine = outcome['update'], \
                     outcome['remove'], outcome['fine'], outcome['spm_fine']
             else:
-                updates, remove, fine, spm_fine = outcome
+                update, remove, fine, spm_fine = outcome
 
         except SystemDatabaseError as err:
             self.error(ERROR_DEP_RESOLUTION_FAILED,
