--- source/Makefile.orig	2010-12-26 14:12:25.298969471 +0100
+++ source/Makefile	2010-12-26 14:13:04.283648054 +0100
@@ -474,16 +474,16 @@
 
 all: client openal qf ded cgame game ui irc angelwrap tv_server
 
-client: $(BUILDDIRS) message-client compile-client link-client start-script-client
+client: $(BUILDDIRS) message-client compile-client link-client
 openal: $(BUILDDIRS) message-openal compile-openal link-openal
 qf: $(BUILDDIRS) message-qf compile-qf link-qf
-ded: $(BUILDDIRS) message-ded compile-ded link-ded start-script-ded
+ded: $(BUILDDIRS) message-ded compile-ded link-ded
 cgame: $(BUILDDIRS) message-cgame compile-cgame link-cgame
 game: $(BUILDDIRS) message-game compile-game link-game
 ui: $(BUILDDIRS) message-ui compile-ui link-ui
 irc: $(BUILDDIRS) message-irc compile-irc link-irc
 angelwrap: $(BUILDDIRS) message-angelwrap compile-angelwrap link-angelwrap
-tv_server: $(BUILDDIRS) message-tv_server compile-tv_server link-tv_server start-script-tv_server
+tv_server: $(BUILDDIRS) message-tv_server compile-tv_server link-tv_server
 
 clean: clean-msg clean-depend clean-client clean-openal clean-qf clean-ded clean-cgame clean-game clean-ui clean-irc clean-angelwrap clean-tv_server
 
