--- source/Makefile.orig	2010-12-26 13:56:44.219938216 +0100
+++ source/Makefile	2010-12-26 13:59:50.813375210 +0100
@@ -48,9 +48,9 @@
 else
 CC?=gcc
 CXX?=g++
-LD=gcc
-AR=ar
-RANLIB=ranlib
+LD=$(CC)
+AR?=ar
+RANLIB?=ranlib
 
 # this nice line comes from the linux kernel makefile
 BASE_ARCH:=$(shell uname -m | sed -e s/i.86/i386/ -e s/sun4u/sparc/ -e s/sparc64/sparc/ -e s/arm.*/arm/ -e s/sa110/arm/ -e s/alpha/axp/)
@@ -166,9 +166,9 @@
 LOCALBASE?=/usr/local
 X11BASE?=/usr/X11R6
 
-CFLAGS_COMMON=$(CFLAGS) -pipe -I. -I$(LOCALBASE)/include -I$(X11BASE)/include -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -Wall -Wno-unused-function
-CFLAGS_RELEASE=-O2 -fno-strict-aliasing -ffast-math -funroll-loops -DNDEBUG
-CFLAGS_DEBUG=-O0 -ggdb -D_DEBUG
+CFLAGS_COMMON=$(CFLAGS) -I. -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -Wall -Wno-unused-function -fno-strict-aliasing
+CFLAGS_RELEASE=-DNDEBUG
+CFLAGS_DEBUG=-D_DEBUG
 ifeq ($(DEBUG_BUILD),YES)
 CFLAGS_COMMON+=$(CFLAGS_DEBUG)
 else
@@ -182,12 +182,9 @@
 
 LDFLAGS_COMMON=$(LDFLAGS)
 ifeq ($(OS),FreeBSD)
-LDFLAGS_COMMON+= -L/usr/local/lib -lm -pthread
+LDFLAGS_COMMON+= -lm -pthread
 else
-LDFLAGS_COMMON+= -L/usr/local/lib -ldl -lm -Wl -O1 --as-needed
-endif
-ifneq ($(DEBUG_BUILD),YES)
-LDFLAGS_COMMON+= -s
+LDFLAGS_COMMON+= -ldl -lm
 endif
 
 ifeq ($(ARCH),x86_64)
