diff -Nur nanny-2.31.1/configure nanny-2.31.1.new/configure
--- nanny-2.31.1/configure	2011-02-07 10:21:40.000000000 +0100
+++ nanny-2.31.1.new/configure	2013-09-13 16:28:30.006570226 +0200
@@ -12026,7 +12026,7 @@
 prog="
 import sys
 try:
-  import Image
+  from PIL import Image
 except RuntimeError:
   sys.exit(0)
 except ImportError:
diff -Nur nanny-2.31.1/configure.ac nanny-2.31.1.new/configure.ac
--- nanny-2.31.1/configure.ac	2011-02-07 10:21:22.000000000 +0100
+++ nanny-2.31.1.new/configure.ac	2013-09-13 16:28:35.126570084 +0200
@@ -104,7 +104,7 @@
 prog="
 import sys
 try:
-  import Image
+  from PIL import Image
 except RuntimeError:
   sys.exit(0)
 except ImportError:
diff -Nur nanny-2.31.1/daemon/src/proxy/TwistedProxy.py nanny-2.31.1.new/daemon/src/proxy/TwistedProxy.py
--- nanny-2.31.1/daemon/src/proxy/TwistedProxy.py	2011-01-23 17:57:16.000000000 +0100
+++ nanny-2.31.1.new/daemon/src/proxy/TwistedProxy.py	2013-09-13 16:28:44.786569817 +0200
@@ -35,7 +35,7 @@
 from tempfile import TemporaryFile, gettempdir
 import time
 
-import Image, ImageDraw, ImageFilter
+from PIL import Image, ImageDraw, ImageFilter
 
 from Controllers import BlockingDeferred
 
