--- mate-panel-1.2.1.orig/applets/fish/Makefile.am
+++ mate-panel-1.2.1/applets/fish/Makefile.am
@@ -31,9 +31,9 @@ libfish_applet_la_LDFLAGS = -module -avo
 libfish_applet_la_CFLAGS = $(AM_CFLAGS)
 else
 APPLET_IN_PROCESS = false
-APPLET_LOCATION = $(libexecdir)/fish-applet
+APPLET_LOCATION = $(libexecdir)/mate-fish-applet
 
-libexec_PROGRAMS = fish-applet
+libexec_PROGRAMS = mate-fish-applet
 fish_applet_SOURCES = $(FISH_SOURCES)
 fish_applet_LDADD = $(FISH_LDADD)
 fish_applet_CFLAGS = $(AM_CFLAGS)

