Remove useless check.
slawomir.nizio at sabayon.org
--- a/configure
+++ b/configure
@@ -256,23 +256,7 @@ fi
 
 # ---- GNOME_CONTROL_CENTER ---- #
 
-if [ "$GNOME_CONTROL_CENTER" = 'Y' ]; then
-  $ECHO -n "... testing Gnome Control Center    : "
-  GNOME_CONTROL_CENTER_DIR=`pkg-config --variable=defappsdir --define-variable=datadir='$(DATADIR)' gnome-default-applications`
-  if [ $? != 0 ]; then
-    if [ $GNOME_CONTROL_CENTER_SPC -eq 1 ]; then
-      ERR "not found. Please install gnome-control-center-dev package. ABORT!\n"
-      $ECHO ""
-      exit 1
-    fi
-    WARN "not found.\n"
-    GNOME_CONTROL_CENTER=N
-  else
-    INFO "found.\n"
-  fi
-fi
-
-$ECHO ""
+GNOME_CONTROL_CENTER_DIR=$DATADIR/gnome-control-center/default-apps
 
 # ---- REPORT ---- #
 
