--- cmake/FindGraphViz.cmake~	2010-11-12 16:07:15.000000000 +0800
+++ cmake/FindGraphViz.cmake	2013-06-02 12:05:28.350114170 +0800
@@ -29,7 +29,7 @@
 if ( NOT WIN32 )
   
   find_package(PkgConfig)
-  pkg_check_modules( graphviz ${REQUIRED} libgvc libcdt libgraph libpathplan )
+  pkg_check_modules( graphviz ${REQUIRED} libgvc libcdt libcgraph libpathplan )
   if ( graphviz_FOUND )
     set ( graphviz_INCLUDE_DIRECTORIES ${graphviz_INCLUDE_DIRS} )
   endif ( graphviz_FOUND )
@@ -66,7 +66,7 @@
     )
 
 find_library( graphviz_GRAPH_LIBRARY
-    NAMES graph
+    NAMES cgraph
     PATHS
     ${graphviz_LIBRARY_DIRS}
     /usr/local/lib64
@@ -162,4 +162,4 @@
 
 # handle the QUIETLY and REQUIRED arguments and set GRAPHVIZ_FOUND to TRUE if 
 # all listed variables are TRUE
-find_package_handle_standard_args(GraphViz DEFAULT_MSG graphviz_LIBRARIES graphviz_INCLUDE_DIRECTORIES)
\ No newline at end of file
+find_package_handle_standard_args(GraphViz DEFAULT_MSG graphviz_LIBRARIES graphviz_INCLUDE_DIRECTORIES)
