--- src/ffmpeg.imageio/~ffmpeginput.cpp	2016-02-20 20:04:19.296006996 +0100
+++ src/ffmpeg.imageio/ffmpeginput.cpp	2016-02-20 20:06:28.888842895 +0100
@@ -207,8 +207,8 @@
         m_offset_time = false;
     }
     AVStream *stream = m_format_context->streams[m_video_stream];
-    if (stream->r_frame_rate.num != 0 && stream->r_frame_rate.den != 0) {
-        m_frame_rate = stream->r_frame_rate;
+    if (stream->avg_frame_rate.num != 0 && stream->avg_frame_rate.den != 0) {
+        m_frame_rate = stream->avg_frame_rate;
     }
     if (static_cast<int64_t>(m_format_context->duration) != int64_t(AV_NOPTS_VALUE)) {
         m_frames = static_cast<uint64_t> ((fps() * static_cast<double>(m_format_context->duration) / 
