Gentoo bug 434472
--- src/CMakeLists.txt.orig
+++ src/CMakeLists.txt
@@ -133,7 +133,7 @@ endif (NVT_TIMEOUT)
 set_target_properties (openvassd PROPERTIES LINK_FLAGS
                        "${LIB_TEMP} ${GLIB_LDFLAGS} ${OPENVAS_LDFLAGS}")
 
-target_link_libraries (openvassd gnutls dl gcrypt)
+target_link_libraries (openvassd gnutls dl gcrypt openvas_base openvas_misc openvas_hg glib-2.0)
 
 set_target_properties (openvassd PROPERTIES COMPILE_FLAGS
                        "${HEADER_TEMP} ${OPENVAS_CFLAGS} ${GLIB_CFLAGS}")
