Index: trunk/client/gui/Makefile.am
===================================================================
--- trunk/client/gui/Makefile.am	(revision 1913)
+++ trunk/client/gui/Makefile.am	(revision 1914)
@@ -69,11 +69,14 @@
 			mv tmp $(SVN_REV_FILE); \
 			sed s/exported/0/ $(SVN_REV_FILE) > tmp; \
 			mv tmp $(SVN_REV_FILE); \
+			sed s/'Unversioned directory'/0/ $(SVN_REV_FILE) > tmp; \
+			mv tmp $(SVN_REV_FILE); \
 			sed s/M.*// $(SVN_REV_FILE) > tmp; \
 			mv tmp $(SVN_REV_FILE);; \
 		*) \
 			sed -i.backup s/:.*// $(SVN_REV_FILE); \
 			sed -i.backup s/exported/0/ $(SVN_REV_FILE); \
+			sed -i.backup s/'Unversioned directory'/0/ $(SVN_REV_FILE); \
 			sed -i.backup s/M.*// $(SVN_REV_FILE);; \
 	esac
 
Index: trunk/client/gui_qt/update_version.sh
===================================================================
--- trunk/client/gui_qt/update_version.sh	(revision 1914)
+++ trunk/client/gui_qt/update_version.sh	(revision 1915)
@@ -20,12 +20,15 @@
         mv tmp $SVN_REV_FILE
         sed s/exported/0/ $SVN_REV_FILE > tmp
         mv tmp $SVN_REV_FILE
+        sed s/'Unversioned directory'/0/ $SVN_REV_FILE > tmp
+        mv tmp $SVN_REV_FILE
         sed s/M.*// $SVN_REV_FILE > tmp
         mv tmp $SVN_REV_FILE
         ;;
     *)
         sed -i.backup s/:.*// $SVN_REV_FILE
         sed -i.backup s/exported/0/ $SVN_REV_FILE
+        sed -i.backup s/'Unversioned directory'/0/ $SVN_REV_FILE
         sed -i.backup s/M.*// $SVN_REV_FILE
         ;;
 esac
