--- a/python/report/plugins/bugzilla/__init__.py
+++ b/python/report/plugins/bugzilla/__init__.py
@@ -115,8 +115,6 @@ import filer
 def sendToBugzilla(component, signature, io, bzfiler,
                    optionsDict, fileName, fileDescription):
 
-    import rpmUtils.arch
-
     class BugzillaCommunicationException (Exception):
         pass
 
@@ -298,7 +296,6 @@ def sendToBugzilla(component, signature,
                     product=bzfiler.getproduct(),
                     component=component,
                     version=bzfiler.getversion(),
-                    platform=rpmUtils.arch.getBaseArch(),
                     bug_severity="medium",
                     priority="medium",
                     op_sys="Linux",
