--- a/etc/Makefile.am
+++ b/etc/Makefile.am
@@ -1,11 +1,9 @@
 nobase_dist_sysconf_DATA = \
-report.d/bugzilla.redhat.com.conf \
-report.d/RHEL-bugzilla.redhat.com.conf \
+report.d/bugs.sabayon.org.conf \
 report.conf \
 report.d/ftp.conf \
 report.d/scp.conf \
 report.d/localsave.conf \
-report.d/strata-test.conf \
-report.d/RHEL.conf
+report.d/strata-test.conf
 
 dist_man_MANS = report.conf.5
--- a/etc/report.d/bugs.sabayon.org.conf
+++ b/etc/report.d/bugs.sabayon.org.conf
@@ -0,0 +1,7 @@
+[bugs.sabayon.org]
+# plugin
+plugin=bugzilla
+
+# rhbz url
+bugzilla_host=bugs.sabayon.org
+
--- a/python/report/plugins/bugzilla/__init__.py
+++ b/python/report/plugins/bugzilla/__init__.py
@@ -1,5 +1,5 @@
 """
-    A Report plugin to send a report to bugzilla.redhat.com.
+    A Report plugin to send a report to bugs.sabayon.org.
     Copyright (C) 2009 Red Hat, Inc
 
     Author(s): Gavin Romig-Koch <gavin@redhat.com>
@@ -54,7 +54,7 @@ def displayURL(optionsDict):
 def bugURL(optionsDict):
     if 'bugURL' in optionsDict:
         return optionsDict["bugURL"]
-    host = "bugzilla.redhat.com"
+    host = "bugs.sabayon.org"
     if 'bugzilla_host' in optionsDict:
         host = optionsDict["bugzilla_host"]
     return "https://" + host + "/xmlrpc.cgi"
@@ -149,8 +149,8 @@ def sendToBugzilla(component, signature,
             else:
                 version = filer.getVersion()
 
-            bzfiler = filer.BugzillaFiler("https://bugzilla.redhat.com/xmlrpc.cgi",
-                                          "http://bugzilla.redhat.com",
+            bzfiler = filer.BugzillaFiler("https://bugs.sabayon.org/xmlrpc.cgi",
+                                          "http://bugs.sabayon.org",
                                           version, product)
 
         if not bzfiler or not bzfiler.supportsFiling() or not bzfiler.bugUrl:
