From bf5ca70551e87671e84a81d103db32ed6918a109 Mon Sep 17 00:00:00 2001
From: Alexander Graf <agraf@suse.de>
Date: Mon, 5 Dec 2011 23:37:52 +0100
Subject: [PATCH 15/33] XXX move qemu binary lower in address space so we have space for guest stuff

---
 x86_64.ld |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/x86_64.ld b/x86_64.ld
index b7a9f4e..1151d8c 100644
--- a/x86_64.ld
+++ b/x86_64.ld
@@ -5,7 +5,7 @@ ENTRY(_start)
 SECTIONS
 {
   /* Read-only sections, merged into text segment: */
-  . = 0x60000000 + SIZEOF_HEADERS;
+  . = 0x8000000 + SIZEOF_HEADERS;
   .interp         : { *(.interp) }
   .hash           : { *(.hash) }
   .dynsym         : { *(.dynsym) }
-- 
1.6.0.2

