From f2bd85fd0ebe444677f22e28ab12b966937207e2 Mon Sep 17 00:00:00 2001
From: Alexander Graf <agraf@suse.de>
Date: Thu, 8 Dec 2011 23:19:32 +0100
Subject: [PATCH 17/33] XXX fake /proc/self/maps: also fclose real file

---
 linux-user/syscall.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/linux-user/syscall.c b/linux-user/syscall.c
index 97c3303..9fabcba 100644
--- a/linux-user/syscall.c
+++ b/linux-user/syscall.c
@@ -4643,6 +4643,7 @@ static int open_self_maps(void *cpu_env, int fd)
     }
 
     free(line);
+    fclose(fp);
 
 #if defined(TARGET_ARM) || defined(TARGET_M68K) || defined(TARGET_UNICORE32)
     dprintf(fd, "%08llx-%08llx rw-p %08llx 00:00 0          [stack]\n",
-- 
1.6.0.2

