diff -ur gscan2pdf-1.1.0/lib/Gscan2pdf/Tesseract.pm gscan2pdf-1.1.0.new/lib/Gscan2pdf/Tesseract.pm
--- gscan2pdf-1.1.0/lib/Gscan2pdf/Tesseract.pm	2012-11-11 21:25:02.000000000 +0100
+++ gscan2pdf-1.1.0.new/lib/Gscan2pdf/Tesseract.pm	2012-11-30 11:57:08.867965437 +0100
@@ -22,8 +22,8 @@
  else {
   return;
  }
- my ( $out, undef ) = Gscan2pdf::Document::open_three("tesseract '' '' -l ''");
- ( $tessdata, $version, $datasuffix ) = parse_tessdata($out);
+ my ( $out, $err ) = Gscan2pdf::Document::open_three("tesseract '' '' -l ''");
+ ( $tessdata, $version, $datasuffix ) = parse_tessdata($err . "\n" . $out);
 
  unless ( defined $tessdata ) {
   if ( defined($version) and $version > 3.01 ) {
