diff -ru opal-3.10.10-old/plugins/video/common/dyna.h opal-3.10.10/plugins/video/common/dyna.h
--- opal-3.10.10-old/plugins/video/common/dyna.h	2013-02-20 03:18:05.000000000 +0100
+++ opal-3.10.10/plugins/video/common/dyna.h	2013-03-18 19:16:14.846866521 +0100
@@ -40,7 +40,8 @@
 
 #ifndef __DYNA_H__
 #define __DYNA_H__ 1
-
+#include <cstdio>
+#include <cstdarg>
 #include "ffmpeg.h"
 #include "critsect.h"
 #include <codec/opalplugin.hpp>
