diff -ru opal-3.10.10-old-p/plugins/video/H.264/h264-x264.cxx opal-3.10.10/plugins/video/H.264/h264-x264.cxx
--- opal-3.10.10-old-p/plugins/video/H.264/h264-x264.cxx	2013-03-18 19:27:45.454835731 +0100
+++ opal-3.10.10/plugins/video/H.264/h264-x264.cxx	2013-03-18 19:32:21.747823413 +0100
@@ -44,6 +44,7 @@
 
 #include "../common/ffmpeg.h"
 #include "../common/dyna.h"
+#define CODEC_FLAG2_DROP_FRAME_TIMECODE 0x00002000
 
 #include "shared/h264frame.h"
 #include "shared/x264wrap.h"
