--- openafs-1.6.1.orig/src/afsd/Makefile.in
+++ openafs-1.6.1/src/afsd/Makefile.in
@@ -34,7 +34,7 @@ afsd: afsd.o afsd_kernel.o $(AFSLIBS) $(
 	${CC} ${CFLAGS} -o afsd afsd.o afsd_kernel.o $(NON_SHARED) $(LDFLAGS) $(AFSD_LDFLAGS) $(AFSLIBS) ${XLIBS} ${AFSD_LIBS}
 
 afsd.fuse: afsd_fuse.o $(UAFSLIBS) $(AFSD_LIBS)
-	${CC} ${UAFS_CFLAGS} -o afsd.fuse afsd_fuse.o $(NON_SHARED) $(LDFLAGS) $(AFSD_LDFLAGS) $(UAFSLIBS) ${UAFS_XLIBS} ${AFSD_LIBS}
+	${CC} ${UAFS_CFLAGS} -pthread -o afsd.fuse afsd_fuse.o $(NON_SHARED) $(LDFLAGS) $(AFSD_LDFLAGS) $(UAFSLIBS) ${UAFS_XLIBS} ${AFSD_LIBS}
 
 vsys: vsys.o
 	${CC} ${CFLAGS} -o vsys vsys.o ${TOP_LIBDIR}/libsys.a $(LDFLAGS) ${XLIBS}
