--- configure.ac.old	2016-02-21 23:27:42.255372985 +0100
+++ configure.ac	2016-02-21 23:33:05.716284195 +0100
@@ -39,7 +39,7 @@ AC_HEADER_DIRENT
 AC_HEADER_STDC
 AC_CHECK_HEADERS([stdlib.h string.h sys/timeb.h])
 
-AC_CHECK_HEADERS([GLee.h], [], AC_MSG_ERROR([Glee is missing.]))
+AC_CHECK_HEADERS([GL/GLee.h], [], AC_MSG_ERROR([Glee is missing.]))
 AC_CHECK_HEADERS([boost/cstdint.hpp], [], AC_MSG_ERROR([Boost is missing.]))
 AC_CHECK_HEADERS([openvdb/openvdb.h], [], AC_MSG_ERROR([OpenVDB is missing.]))

