--- src/Makefile.orig	2013-02-21 16:33:04.872791037 +0000
+++ src/Makefile	2013-02-21 16:34:21.900019906 +0000
@@ -2441,10 +2441,13 @@
 .c.o:
 	$(CCC) $<
 
+XSUBPP = `$(PERL) -MModule::Build -e'print Module::Build::ModuleInfo->find_module_by_name("ExtUtils::xsubpp")'`
+TYPEMAP = `$(PERL) -MModule::Build -e'print Module::Build::ModuleInfo->find_module_by_name("ExtUtils::typemap")'`
+
 auto/if_perl.c: if_perl.xs
 	$(PERL) -e 'unless ( $$] >= 5.005 ) { for (qw(na defgv errgv)) { print "#define PL_$$_ $$_\n" }}' > $@
-	$(PERL) $(PERLLIB)/ExtUtils/xsubpp -prototypes -typemap \
-	    $(PERLLIB)/ExtUtils/typemap if_perl.xs >> $@
+	$(PERL) $(XSUBPP) -prototypes -typemap \
+	    $(TYPEMAP) if_perl.xs >> $@
 
 auto/osdef.h: auto/config.h osdef.sh osdef1.h.in osdef2.h.in
 	CC="$(CC) $(OSDEF_CFLAGS)" srcdir=$(srcdir) sh $(srcdir)/osdef.sh