--- Makefile.in.orig	2010-01-04 12:49:38.000000000 -0200
+++ Makefile.in	2010-01-04 12:49:41.000000000 -0200
@@ -792,7 +792,7 @@
 	@for p in tkgate verga; do \
 	   f=`echo "$$p" | sed 's,^.*/,,;$(transform);s/$$/$(EXEEXT)/'`; \
 	   rm -f $(DESTDIR)$(pkgdatadir)/libexec/$$p; \
-	   $(LN_S) `(cd $(DESTDIR)$(bindir);pwd)`/$$f $(DESTDIR)$(pkgdatadir)/libexec/$$p; \
+	   $(LN_S) $(bindir)/$$f $(DESTDIR)$(pkgdatadir)/libexec/$$p; \
 	done
 #
 # Copy all the data files we need
