diff -urN pyload.orig/module/config/default.conf pyload.new/module/config/default.conf
--- pyload.orig/module/config/default.conf	2011-12-29 13:06:43.000000000 +0100
+++ pyload.new/module/config/default.conf	2012-01-13 00:17:20.351549582 +0100
@@ -2,7 +2,7 @@
 
 remote - "Remote":
 	int port : "Port" = 7227
-	ip listenaddr : "Adress" = 0.0.0.0
+	ip listenaddr : "Adress" = 127.0.0.1
 	bool nolocalauth : "No authentication on local connections" = True
 	bool activated : "Activated" = True
 ssl - "SSL":
@@ -13,7 +13,7 @@
 	bool activated : "Activated" = True
 	builtin;threaded;fastcgi;lightweight server : "Server" = builtin
 	bool https : "Use HTTPS" = False
-	ip host : "IP" = 0.0.0.0
+	ip host : "IP" = 127.0.0.1
 	int port : "Port" = 8001
 	str template : "Template" = default
     str prefix: "Path Prefix" =
