diff -Nurp entropy-180.orig/services/kernel-switcher entropy-180/services/kernel-switcher
--- entropy-180.orig/services/kernel-switcher	2013-03-03 01:08:10.000000000 +0000
+++ entropy-180/services/kernel-switcher	2013-03-06 11:54:12.760696615 +0000
@@ -237,7 +237,7 @@ def _switch_kernel(nsargs):
         rc, _show_cfgupd = install._install_action(
             etp_client, True, True,
             pretend, ask, verbose,
-            quiet, False, False, False, False, False,
+            quiet, False, False, False, False, False, False,
             False, 1, [], package_matches=list(matches))
         if _show_cfgupd:
             install._show_config_files_update(etp_client)
