--- configure-old	2014-02-03 18:49:57.524228900 +0100
+++ configure	2014-02-03 18:54:31.920895855 +0100
@@ -13025,13 +13025,13 @@
   MNG_LIBS="${MNG_LIBS} -lz"
 fi
 
-            { $as_echo "$as_me:${as_lineno-$LINENO}: checking for cmsCloseProfile in -llcms" >&5
-$as_echo_n "checking for cmsCloseProfile in -llcms... " >&6; }
+            { $as_echo "$as_me:${as_lineno-$LINENO}: checking for cmsCloseProfile in -llcms2" >&5
+$as_echo_n "checking for cmsCloseProfile in -llcms2... " >&6; }
 if ${ac_cv_lib_lcms_cmsCloseProfile+:} false; then :
   $as_echo_n "(cached) " >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-llcms  $LIBS"
+LIBS="-llcms2  $LIBS"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -13062,7 +13062,7 @@
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_lib_lcms_cmsCloseProfile" >&5
 $as_echo "$ac_cv_lib_lcms_cmsCloseProfile" >&6; }
 if test "x$ac_cv_lib_lcms_cmsCloseProfile" = xyes; then :
-  MNG_LIBS="${MNG_LIBS} -llcms"
+  MNG_LIBS="${MNG_LIBS} -llcms2"
 fi
 
 
--- configure.ac-old	2014-02-03 18:49:17.970895525 +0100
+++ configure.ac	2014-02-03 18:53:08.964229101 +0100
@@ -538,7 +538,7 @@
             MNG_LIBS="${MNG_LIBS} -lmng"
             MNG_LIBS="${MNG_LIBS} ${JPEG_LIBS}"
             AC_CHECK_LIB([z],    [zlibVersion],    [MNG_LIBS="${MNG_LIBS} -lz"   ])
-            AC_CHECK_LIB([lcms], [cmsCloseProfile],[MNG_LIBS="${MNG_LIBS} -llcms"])
+            AC_CHECK_LIB([lcms2], [cmsCloseProfile],[MNG_LIBS="${MNG_LIBS} -llcms2"])
           ],
           [AC_MSG_ERROR(['libmng' library file was not found.])]
         )
