--- iscsitarget-1.4.20.2.orig/kernel/block-io.c
+++ iscsitarget-1.4.20.2/kernel/block-io.c
@@ -9,6 +9,7 @@
  * This code is licenced under the GPL.
  */
 
+#include <linux/module.h>
 #include <linux/types.h>
 #include <linux/blkdev.h>
 #include <linux/parser.h>
--- iscsitarget-1.4.20.2.orig/kernel/config.c
+++ iscsitarget-1.4.20.2/kernel/config.c
@@ -4,6 +4,7 @@
  * This code is licenced under the GPL.
  */
 
+#include <linux/module.h>
 #include <linux/proc_fs.h>
 
 #include "iscsi.h"
--- iscsitarget-1.4.20.2.orig/kernel/event.c
+++ iscsitarget-1.4.20.2/kernel/event.c
@@ -6,6 +6,7 @@
  * Some functions are based on audit code.
  */
 
+#include <linux/module.h>
 #include <net/tcp.h>
 #include "iet_u.h"
 #include "iscsi_dbg.h"
--- iscsitarget-1.4.20.2.orig/kernel/session.c
+++ iscsitarget-1.4.20.2/kernel/session.c
@@ -4,6 +4,8 @@
  * Released under the terms of the GNU GPL v2.0.
  */
 
+#include <linux/module.h>
+
 #include "iscsi.h"
 #include "iscsi_dbg.h"
 
--- iscsitarget-1.4.20.2.orig/kernel/target.c
+++ iscsitarget-1.4.20.2/kernel/target.c
@@ -4,6 +4,7 @@
  * Released under the terms of the GNU GPL v2.0.
  */
 
+#include <linux/module.h>
 #include <linux/mutex.h>
 
 #include "iscsi.h"
--- iscsitarget-1.4.20.2.orig/kernel/volume.c
+++ iscsitarget-1.4.20.2/kernel/volume.c
@@ -4,6 +4,7 @@
  * This code is licenced under the GPL.
  */
 
+#include <linux/module.h>
 #include <linux/types.h>
 #include <linux/parser.h>
 

