--- interface/vmcs_host/linux/vcfiled/vcfiled_check.h
+++ interface/vmcs_host/linux/vcfiled/vcfiled_check.h
@@ -34,8 +34,8 @@
 #endif
 
 #ifndef VCFILED_LOCKFILE
-#define VCFILED_LOCKDIR "/var/run/vcfiled"
-#define VCFILED_LOCKFILE VCFILED_LOCKDIR "/vcfiled"
+#define VCFILED_LOCKDIR "/var/run"
+#define VCFILED_LOCKFILE VCFILED_LOCKDIR "/vcfiled.pid"
 #endif
 
 typedef void (*VCFILED_LOGMSG_T)(int level, const char *fmt, ...);
