--- a/mysql/config/ac-macros/perftools.m4
+++ b/mysql/config/ac-macros/perftools.m4
@@ -87,10 +87,14 @@ case "$mysql_perftools_dir" in
     if test \( -f "$mysql_perftools_dir/lib/libprofiler.a"  -o \
                -f "$mysql_perftools_dir/lib/libprofiler.so" -o \
                -f "$mysql_perftools_dir/lib/libprofiler.sl" -o \
-               -f "$mysql_perftools_dir/lib/libprofiler.dylib" \) \
-            -a -f "$mysql_perftools_dir/include/tcmalloc.h"; then
-      PERFTOOLS_INCLUDES="-I$mysql_perftools_dir/include"
+               -f "$mysql_perftools_dir/lib/libprofiler.dylib" \); then
       PERFTOOLS_LIBS="-L$mysql_perftools_dir/lib"
+      if test \( -f "$mysql_perftools_dir/include/tcmalloc.h" \); then
+        PERFTOOLS_INCLUDES="-I$mysql_perftools_dir/include"
+      fi
+      if test \( -f "$mysql_perftools_dir/include/gperftools/tcmalloc.h" \); then
+        PERFTOOLS_INCLUDES="-I$mysql_perftools_dir/include/gperftools"
+      fi
       GOOGLE_CHECK_PERFTOOLS_DIR
     fi
     if test "x$mysql_cv_perftools" != "xyes"; then 


