diff -urN a/econnman-bin b/econnman-bin
--- a/econnman-bin	2012-08-13 02:17:11.000000000 +0300
+++ b/econnman-bin	2013-04-13 20:54:17.902143514 +0300
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 
 # TODO:
diff -urN a/econnman-bin.in b/econnman-bin.in
--- a/econnman-bin.in	2012-08-04 01:08:14.000000000 +0300
+++ b/econnman-bin.in	2013-04-13 20:54:24.273063866 +0300
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 
 # TODO:
