--- streamtuner2.desktop
+++ streamtuner2.desktop
@@ -1,14 +1,13 @@
 [Desktop Entry]
-Encoding=UTF-8
-Value=1.0
+Version=1.0
 Type=Application
 
 Exec=streamtuner2
-Icon=streamtuner2.png
+Icon=streamtuner2
 
 Name=streamtuner2
 Name[en]=streamtuner2
 GenericName=Internet radio browser
 Comment=Listen to shoutcast/xiph/live365 music stations
 
-Categories=AudioVideo
\ No newline at end of file
+Categories=AudioVideo;
