--- mongodb-src-r2.0.4.orig/SConstruct
+++ mongodb-src-r2.0.4/SConstruct
@@ -711,7 +711,7 @@ if nix:
     env.Append( CXXFLAGS=" -Wnon-virtual-dtor " )
     env.Append( LINKFLAGS=os.environ['LDFLAGS'] )
     env.Append( LINKFLAGS=" -fPIC -pthread -rdynamic" )
-    env.Append( LIBS=[] )
+    env.Append( LIBS=[ "-lpcre" ] )
 
     #make scons colorgcc friendly
     env['ENV']['HOME'] = os.environ['HOME']
