--- a/src/mongo/client/sasl_client_session.cpp
+++ b/src/mongo/client/sasl_client_session.cpp
@@ -29,17 +29,17 @@ namespace {
      * doesn't initialize the library for us.
      */
 
-    void* saslOurMalloc(unsigned long sz) {
+    void* saslOurMalloc(size_t sz) {
         return ourmalloc(sz);
     }
 
-    void* saslOurCalloc(unsigned long count, unsigned long size) {
+    void* saslOurCalloc(size_t count, size_t size) {
         void* ptr = calloc(count, size);
         if (!ptr) printStackAndExit(0);
         return ptr;
     }
 
-    void* saslOurRealloc(void* ptr, unsigned long sz) {
+    void* saslOurRealloc(void* ptr, size_t sz) {
         return ourrealloc(ptr, sz);
     }
 

