diff -ru noad-0.6.0-orig/Makefile.am noad-0.6.0/Makefile.am
--- noad-0.6.0-orig/Makefile.am	2006-06-19 09:40:49.000000000 +0200
+++ noad-0.6.0/Makefile.am	2006-06-19 09:41:47.000000000 +0200
@@ -4,8 +4,8 @@
 libnoad_a_SOURCES=tools.cpp vdr_cl.cpp svdrpc.cpp tools.h vdr_cl.h svdrpc.h mpeg2wrap.cpp mpeg2wrap.h
 
 bin_PROGRAMS=noad
-noad_SOURCES=cchecklogo.cpp ccontrol.cpp cgetlogo.cpp ctoolbox.cpp noad.cpp tnoad.cpp noaddata.cpp\
-	           audiotools.cpp main.cpp \
+noad_SOURCES=main.cpp cchecklogo.cpp ccontrol.cpp cgetlogo.cpp ctoolbox.cpp noad.cpp tnoad.cpp noaddata.cpp\
+	           audiotools.cpp \
 	    	audiotools.h cchecklogo.h ccontrol.h cgetlogo.h ctoolbox.h noad.h tnoad.h noaddata.h videodir.h
 
 noad_LDADD=libnoad.a
@@ -14,8 +14,7 @@
 EXTRA_PROGRAMS=markpics showindex
 
 markpics_SOURCES=markpics.cpp
-markpics_LDADD=libnoad.a
-markpics_LDFLAGS=-lMagick++
+markpics_LDADD=libnoad.a -lMagick++
 
 showindex_SOURCES=showindex.cpp
 showindex_LDADD=libnoad.a
