diff -ru noad-0.6.0-orig/configure.ac noad-0.6.0/configure.ac
--- noad-0.6.0-orig/configure.ac	2006-06-19 09:35:21.005806971 +0200
+++ noad-0.6.0/configure.ac	2006-06-19 09:37:45.574799434 +0200
@@ -122,16 +122,15 @@
 
 AC_SUBST(PICSRC,"")
 AC_MSG_CHECKING(for Magick++(2))
-CFLAGS=$ac_save_CFLAGS
-ac_save_CXXFLAGS="$CXXFLAGS"
-CXXFLAGS="$CXXFLAGS -lMagick++"
+save_LDADD="$LDADD"
+LDADD="$LDADD -lMagick++"
 AC_COMPILE_IFELSE([
 AC_LANG_PROGRAM([#include <Magick++.h>
 using namespace Magick;
 ],[Image image;])
 ],[AC_MSG_RESULT("yes");mpp_defined="1"],[AC_MSG_RESULT("no");mpp_defined="0"]
 )
-CXXFLAGS=$ac_save_CXXFLAGS
+LDADD=$save_LDADD
 test $mpp_defined -eq 1 && test $havemagick -eq 1 && PICSRC="markpics"
 
 dnl check for request to compile toolies
@@ -170,7 +169,7 @@
 AC_CHECK_FUNCS([gettimeofday localtime_r memmove memset mkdir select socket strchr strdup strerror strrchr strstr])
 AC_CHECK_FUNCS([strsignal])
 #CXXFLAGS='-g -O3 -pg'
-CXXFLAGS='-g -O3'
+#CXXFLAGS='-g -O3'
 
 if test "$withffmpeg" == "1"; then 
   dnl check for a usable avcodec.h 
