diff -ru noad-0.6.0-orig/tools.cpp noad-0.6.0/tools.cpp
--- noad-0.6.0-orig/tools.cpp	2006-09-22 14:33:32.000000000 +0200
+++ noad-0.6.0/tools.cpp	2006-10-06 10:09:35.000000000 +0200
@@ -17,6 +17,8 @@
 #include <time.h>
 #include <unistd.h>
 #include <execinfo.h>
+#include <sys/types.h>
+#include <signal.h>
 
 extern int SysLogLevel;
 
@@ -961,6 +963,10 @@
     esyslog(LOG_ERR, "ERROR: frame larger than buffer (%d > %d)", Length, Max);
     Length = Max;
   }
+  if (f == -1) {
+    esyslog(LOG_ERR, "ERROR: cannot open video-file (recording deleted while scanning or bad index-file)");
+    kill(getpid(),SIGTERM);
+  }
   //int r = safe_read(f, b, Length);
   int r = read( f,b,Length);
   if (r < 0)
