diff -ru noad-0.6.0-orig/vdr_cl.h noad-0.6.0/vdr_cl.h
--- noad-0.6.0-orig/vdr_cl.h	2006-10-07 11:25:46.164645112 +0200
+++ noad-0.6.0/vdr_cl.h	2006-10-07 11:26:01.022074938 +0200
@@ -52,7 +52,7 @@
 
 #define KILOBYTE(n) ((n) * 1024)
 // The maximum size of a single frame:
-#define MAXFRAMESIZE  KILOBYTE(192)
+#define MAXFRAMESIZE  KILOBYTE(256)
 
 #define FRAMESPERSEC 25
 #define FRAMESPERMIN (FRAMESPERSEC*60)
