Index: noad-0.6.0/audiotools.cpp
===================================================================
--- noad-0.6.0.orig/audiotools.cpp
+++ noad-0.6.0/audiotools.cpp
@@ -99,7 +99,8 @@ int scan_audio_stream_0(unsigned char *m
   inbuf_ptr = mbuf;
   while (size > 0) 
   {
-    len = avcodec_decode_audio(c, (short *)outbuf, &out_size, 
+    out_size=AVCODEC_MAX_AUDIO_FRAME_SIZE;
+    len = avcodec_decode_audio2(c, (short *)outbuf, &out_size, 
                                   inbuf_ptr, size);
     if (len < 0) 
     {
